src/Test/DepTyCheck/Gen/Coverage.idr

```diff
@@ -41,10 +41,22 @@ MonadWriter ModelCoverage m => CanManageLabels m where
   manageLabel l x = tell (MkModelCoverage $ singleton l 1) >> x
 
 export
-unGenTryAllD : RandomGen g => (seed : g) -> Gen em a -> Stream $ Maybe (ModelCoverage, a)
-unGenTryAllD seed gen = do
+unGenD : MonadRandom m => MonadError () m => CanManageLabels m => Gen em a -> m (ModelCoverage, a)
```

Comment:
Looks like no `CanManageLabels m` is required in this function, because `WriterT` is used to manage labels anyway. I wonder why there is no ambiguity error for `CanManageLabels` in the `unGen` call -- I think, both `WriterT` and external one provide them.

Anyway, there should be a test for this new functions, at least for this basic `unGenD` one, like there are tests in `tests/lib/coverage` for `unGenTryND`, just in case. In case of `CanManageLabels` is left, those tests should test that we indeed get both correct `ModelCoverage` value, and still can indeed manage labels using this external `auto`-implicit argument.